apps/cowswap-frontend/src/modules/orders/pure/PendingOrderNotification/index.tsx

```diff
@@ -105,24 +63,33 @@ export function PendingOrderNotification(props: PendingOrderNotificationProps) {
     },
   }
 
+  const ref = useCallback(
+    (node: HTMLDivElement) => {
+      if (node) onToastMessage(node.innerText)
+    },
+    [onToastMessage]
```

Comment:
Please forgive my ignorance. How does this work?
When will this trigger? When the component is rendered?

Reply:
I'm also confused. Also, there's a bit of repetition. Once we agree if this is a good pattern, if we keep it, I would suggest to build a custom hook

Reply:
@anxolin I've got rid of the code duplications in the last PR: https://github.com/cowprotocol/cowswap/pull/3934

Reply:
>How does this work?

>React will call the ref callback with the DOM element when the component mounts, and call it with null when it unmounts

https://legacy.reactjs.org/docs/refs-and-the-dom.html#callback-refs